config/v1/types_dns.go

```diff
@@ -96,3 +103,31 @@ type DNSList struct {
 
 	Items []DNS `json:"items"`
 }
+
+// DNSPlatformSpec holds cloud-provider-specific configuration
+// for DNS administration.
+// +union
+type DNSPlatformSpec struct {
+	// type is the underlying infrastructure provider for the cluster.
+	// Allowed values are "AWS", "Azure", "BareMetal", "GCP", "Libvirt",
```

Comment:
FYI: when the list of platforms this description becomes outdated. Thus, better to mention which values are currently supported instead for the new struct. E.g. `// Currently supported value is AWS.`. Extend it later once new platform specific fields are added.